pkg/envutil/envutil.go

```diff
@@ -0,0 +1,146 @@
+// SPDX-FileCopyrightText: Copyright The Lima Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package envutil
+
+import (
+	"os"
+	"strings"
+
+	"github.com/sirupsen/logrus"
+)
+
+// BuiltinBlocklist contains environment variables that should not be propagated by default.
+var BuiltinBlocklist = []string{
+	"BASH*",
+	"DISPLAY",
+	"DYLD_*",
+	"EUID",
+	"FPATH",
+	"GID",
+	"GROUP",
+	"HOME",
+	"HOSTNAME",
+	"LD_*",
+	"LOGNAME",
+	"OLDPWD",
+	"PATH",
+	"PWD",
+	"SHELL",
+	"SHLVL",
+	"SSH_*",
+	"TERM",
+	"TERMINFO",
+	"TMPDIR",
+	"UID",
+	"USER",
+	"XAUTHORITY",
+	"XDG_*",
+	"ZDOTDIR",
+	"ZSH*",
+	"_*", // Variables starting with underscore are typically internal
+}
+
+func getBlockList() []string {
+	if blockEnv := os.Getenv("LIMA_SHELLENV_BLOCK"); blockEnv != "" {
+		if strings.HasPrefix(blockEnv, "+") {
+			additionalBlocks := parseEnvList(blockEnv[1:])
+			blockList := make([]string, len(BuiltinBlocklist)+len(additionalBlocks))
+			copy(blockList, BuiltinBlocklist)
+			copy(blockList[len(BuiltinBlocklist):], additionalBlocks)
+			return blockList
+		}
+		return parseEnvList(blockEnv)
+	}
+	return BuiltinBlocklist
+}
```

Comment:
Let's simplify implementation:
```suggestion
func getBlockList() []string {
	blockEnv := os.Getenv("LIMA_SHELLENV_BLOCK")
	if blockEnv == "" {
		return BuiltinBlocklist
	}

	after, found := strings.CutPrefix(blockEnv, "+")
	if !found {
		return parseEnvList(blockEnv)
	}

	return slices.Concat(BuiltinBlocklist, parseEnvList(after))
}
```